go/database/mpt/nodes.go

```diff
@@ -327,7 +330,7 @@ func visitPathTo(source NodeSource, root *NodeReference, path []Nibble, address
 			return false, nil
 		}
 
-		if res := visitor.Visit(last.Get(), NodeInfo{Id: lastNodeId.Id()}); res != VisitResponseContinue {
+		if res := visitor.Visit(last.Get(), NodeInfo{Id: lastNodeId.Id(), Embedded: nextEmbedded && found}); res != VisitResponseContinue {
```

Comment:
Why is `nextEmbedded` combined with `found`? 

It is very hard for me to think about the semantic of this flag. If I write a Visitor, the NodeInfo`s "Embedded" field may lie to me?

Reply:
Basically I re-used this flag to forward the `embedded` flag to next iteration. At some point a node (must be branch or ext) tells you that the next node will be embedded.  You have to save this information for the next loop to pass it to the visitor.  I used the `found` flag for it, assuming that the only option for the next iteration is that it must reach a correct value node, which will set the `found` flag to true.

As you write in the other comment, if you iterated to another value node (or even a different type of node), `found` would be false and the information passed to the visitor would be incorrect - but this is a situation which I guess cannot happen, i.e. I did not considered it. 

If you like, I can use one more bool flag serving only for the embedded flag, it may be more readable, I see that.

Reply:
If you can improve the readability, please do so. Right now, it is very hard to wrap ones head around it.

Reply:
done, I introduced a new flag that is passed through. 

also I could create a test that brakes the previous version, i.e. it was not entirely correct assumption. .